static/app/components/modals/inviteMembersModal/inviteMembersModalNew.tsx

```diff
@@ -0,0 +1,168 @@
+import InviteStatusMessage from 'sentry/components/modals/inviteMembersModal/inviteStatusMessage';
+import type {
+  InviteRow,
+  InviteStatus,
+  NormalizedInvite,
+} from 'sentry/components/modals/inviteMembersModal/types';
+import {ORG_ROLES} from 'sentry/constants';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Member} from 'sentry/types/organization';
+
+interface Props {
+  Body: ModalRenderProps['Body'];
+  Footer: ModalRenderProps['Footer'];
+  Header: ModalRenderProps['Header'];
+  canSend: boolean;
+  complete: boolean;
+  headerInfo: ReactNode;
+  inviteStatus: InviteStatus;
+  invites: NormalizedInvite[];
+  member: Member | undefined;
+  pendingInvites: InviteRow[];
+  reset: () => void;
+  sendInvites: () => void;
+  sendingInvites: boolean;
+  setEmails: (emails: string[], index: number) => void;
+  setRole: (role: string, index: number) => void;
+  setTeams: (teams: string[], index: number) => void;
+  willInvite: boolean;
+  error?: string;
+}
+
+export default function InviteMembersModalNew({
+  canSend,
+  complete,
+  Header,
+  Body,
+  Footer,
+  headerInfo,
+  invites,
+  inviteStatus,
+  member,
+  pendingInvites,
+  reset,
+  sendingInvites,
+  sendInvites,
+  setEmails,
+  setRole,
+  setTeams,
+  willInvite,
+  error,
+}: Props) {
+  const isValidInvites = invites.length > 0;
+  const {emails, role, teams} = pendingInvites[0] ?? {};
+
+  const errorAlert = error ? (
+    <Alert type="error" showIcon>
+      {error}
+    </Alert>
+  ) : null;
+
+  return (
+    <Fragment>
+      <Header closeButton>
+        {errorAlert}
+        <Heading>{t('Invite New Members')}</Heading>
+      </Header>
+      <Body>
+        {willInvite ? (
+          <Subtext>
+            {t('Invite unlimited new members to join your organization.')}
+          </Subtext>
+        ) : (
+          <Alert type="warning" showIcon>
+            {t(
+              'You can’t invite users directly, but we’ll forward your request to an org owner or manager for approval.'
+            )}
+          </Alert>
+        )}
+
+        {headerInfo}
+
+        <Rows>
+          <StyledInviteRow
+            disabled={false}
+            emails={[...emails]}
+            role={role}
+            teams={[...teams]}
+            roleOptions={member?.orgRoleList ?? ORG_ROLES}
+            roleDisabledUnallowed={willInvite}
+            inviteStatus={inviteStatus}
+            onRemove={reset}
+            onChangeEmails={opts => setEmails(opts?.map(v => v.value) ?? [], 0)}
+            onChangeRole={value => setRole(value?.value, 0)}
+            onChangeTeams={opts => setTeams(opts ? opts.map(v => v.value) : [], 0)}
+          />
+        </Rows>
+
+        <Footer>
+          <FooterContent>
+            <div>
+              <InviteStatusMessage
+                complete={complete}
+                hasDuplicateEmails={false}
```

Comment:
We can remove this `hasDuplicateEmails` prop once we stop using the old invite modal, since there will no longer be multiple invite rows and thus no possibility for duplicate emails.